lmdeploy/pytorch/paging/scheduler.py

```diff
@@ -212,17 +206,15 @@ def _deactive_adapter(adapter_name):
                 if seq.adapter_name not in required_adapters:
                     break
 
-            if not block_manager.can_allocate(seq):
-                if not _evict_until_can_append(seq, waiting):
-                    break
+            self.block_trie.match(seq)
+
+            if not __evict_for_seq(seq, waiting):
+                break
 
-            if eviction_helper.need_swap_in(seq):
-                if not eviction_helper.try_swap_in(seq, swap_in_map):
-                    break
             # allocate session memory
-            block_manager.allocate(seq)
+            self.block_manager.allocate(seq)
+            self.block_trie.allocate(seq)
```

Comment:
I think we can remove this `block_trie.allocate` here, since there is no computed blocks to cache before the prefill step.

Reply:
Blocks can be shared even when they have not finished.

Reply:
I am concerning if there are some write conflicts for these unfinished blocks. If we only take computed blocks, they are read-only and safe.